samples/azure-quantum/shorter-quantum-circuits/shorter-quantum-circuits-dataset.ipynb

```diff
@@ -0,0 +1,782 @@
+    "Each JSON file contains the result of approximating one diagonal rotation gate by a fixed target angle \n",
+    "using one of the four approximation protocols. \n",
+    "The name of each JSON file starts with a prefix representing the rotation angle using scientific notation, \n",
+    "and ends with a suffix corresponding to the approximation protocol:\n",
+    "\n",
+    "* `diagonal.json` for diagonal approximation \n",
+    "* `fallback.json` for fallback approximation\n",
+    "* `mixed-diagonal.json` for mixed diagonal approximation \n",
+    "* `mixed-fallback.json` mixed fallback approximation\n",
+    "\n",
+    "The naming convention for the JSON files is `<angle>-<protocol-kind>.json`. Each file contains a series of descriptions of the corresponding protocol with increasing diamond norm accuracy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tarfile\n",
+    "import itertools\n",
+    "with tarfile.open('clifford-root-t-fourier-angles.tar.gz') as tar:\n",
+    "    for member in itertools.islice(tar,10):\n",
+    "        print(member.name)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### File structure\n",
+    "\n",
+    "Each file contains an array $a_k$ of the best found approximation protocols that use a denominator power of at most $k$ in the branch that is applied with the highest probability. Each element of the array corresponds to one instance of the approximation protocol.\n",
+    "\n",
+    "Each protocol instance description in the JSON file contains three keys:\n",
+    "\n",
+    "* `target_angle` for the target rotation angle\n",
+    "* `protocol_name` for the protocol name\n",
+    "* `elements` for a description of the gate sequences and the probabilities with which these sequences are applied."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tarfile\n",
+    "import json\n",
+    "with tarfile.open('clifford-root-t-fourier-angles.tar.gz') as tar:\n",
+    "    protocol_instances = json.loads(tar.extractfile(tar.getmembers()[2]).read().decode())\n",
+    "    print(type(protocol_instances), ' of length ', len(protocol_instances))\n",
+    "    print(\"Example of a protocol instance:\\n\", json.dumps(protocol_instances[-1],indent=2))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Protocol instance structure\n",
+    "\n",
+    "The array corresponding to the key `element` of the protocol instance data structure is an array of length one, two, four or six. \n",
+    "Each array item contains three keys:\n",
+    "* `gates` - a string representing a product of unitary matrices $V$, for example `Rz[1,16].H.S` represents matrix product $V = \\exp(-\\frac{i\\pi}{16}Z)HS$\n",
```

Comment:
It's not obvious where the `1` inside `Rz[1,16]` is used in the matrix product. Could you make it explicit?